inet/vibe/inet/url.d

```diff
@@ -19,11 +19,43 @@ import std.exception;
 import std.string;
 import std.traits : isInstanceOf;
 import std.ascii : isAlpha, isASCII, toLower;
-import std.uri: encode;
+import std.uri: decode, encode;
 
 import core.checkedint : addu;
 
 
+/** Parses a user-provided URL with relaxed rules.
+
+	Unlike `URL.parse`, this allows the URL to use special characters as part of
+	the host name and path, automatically employing puny code or percent-encoding
+	to convert this to a valid URL.
+
+	Params:
+		url = String representation of the URL
+*/
+URL parseUserURL(string url)
```

Comment:
Okay, I now see that essentially, you just wanted to tidy up the API a bit ?
I don't think `parseUserURL` is very useful in its current incarnation though: future changes might need to add parameters, so that'll be a breaking change, which means we'll probably deprecate this overload to add a `URL parseUserURL(string url, string default_protocol)` sooner than later. In that regard, keeping `parsePlain` doesn't seem to bad to me.

But if you prefer to go with that approach, it won't make a huge difference when the time to improve `parseUserURL` comes, so feel free to merge as-is. I'm quite eager to be able to compile Vibe.d on my machine again :D

Reply:
Agreed, we'll probably add it like that later anyway. I'll add the parameter now then and ignore it until the logic is implemented (still some time until the final 0.9.5 release is tagged).

Reply:
> In that regard, keeping parsePlain doesn't seem to bad to me.

IMO the word "plain" just isn't right here and I also wanted to separate the code from the struct definition, because I'd expect that the implementation will become considerably more complex in the future.